datafusion/core/src/datasource/physical_plan/file_scan_config.rs

```diff
@@ -445,30 +410,7 @@ impl PartitionColumnProjector {
                         "Invalid partitioning found on disk".to_string(),
                     ))?;
 
-            let mut partition_value = Cow::Borrowed(p_value);
-
-            // check if user forgot to dict-encode the partition value
-            let field = self.projected_schema.field(sidx);
```

Comment:
I check the `field.data_type()`, and only get the value type but not Dictionary. I'm not sure why, but I expect Dict is preserved at this point 🤔

Reply:
This should be fixed now as I use the schema information.